mlflow/utils/autologging_utils.py

```diff
@@ -70,39 +36,27 @@ def log_fn_args_as_params(fn, args, kwargs, unlogged=[]):  # pylint: disable=W01
     :param unlogged: parameters not to be logged
     :return: None
     """
-    # all_default_values has length n, corresponding to values of the
-    # last n elements in all_param_names
-    pos_params, _, _, pos_defaults, kw_params, kw_defaults, _ = inspect.getfullargspec(fn)
-
-    kw_params = list(kw_params) if kw_params else []
-    pos_defaults = list(pos_defaults) if pos_defaults else []
-    all_param_names = pos_params + kw_params
-    all_default_values = pos_defaults + [kw_defaults[param] for param in kw_params]
-
-    # Checking if default values are present for logging. Known bug that getargspec will return an
-    # empty argspec for certain functions, despite the functions having an argspec.
-    if all_default_values is not None and len(all_default_values) > 0:
-        # Logging the default arguments not passed by the user
-        defaults = get_unspecified_default_args(args, kwargs, all_param_names, all_default_values)
-
-        for name in [name for name in defaults.keys() if name in unlogged]:
-            del defaults[name]
-        try_mlflow_log(mlflow.log_params, defaults)
-
-    # Logging the arguments passed by the user
-    args_dict = dict(
-        (param_name, param_val)
-        for param_name, param_val in zip(all_param_names, args)
-        if param_name not in unlogged
+    param_spec = inspect.signature(fn).parameters
+    # Fetch the parameter names for specified positional arguments from the function
+    # signature & create a mapping from positional argument name to specified value
+    params_to_log = {
+        param_info.name: param_val
+        for param_info, param_val in zip(list(param_spec.values())[len(args) :], args)
+    }
```

Comment:
```suggestion
    params_to_log = {
        param_info.name: param_val
        for param_info, param_val in zip(list(param_spec.values()), args)
    }
```

Should `[len(args) :]` here be removed?

Reply:
I don't think so, unless I'm missing something. `param_spec.values()` contains all of the methods parameters in order: every positional argument and keyword argument. To find parameter names for only the user-specified `args`, we should slice the first `len(args)` arguments, which are guaranteed to be positional arguments (or positional-or-keyword type arguments). Let me know if this makes sense.

Reply:
Ah, this should be `:len(args)` though.

Reply:
Fixed and added a test case: `test_tf_keras_autolog_names_positional_parameters_correctly`